src/LevelDBStore/DB.cs

```diff
@@ -0,0 +1,459 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// DB.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using LevelDB.NativePointer;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace LevelDB
+{
+    /// <summary>
+    /// A DB is a persistent ordered map from keys to values.
+    /// A DB is safe for concurrent access from multiple threads without any external synchronization.
+    /// </summary>
+    public class DB : LevelDBHandle, IEnumerable<KeyValuePair<string, string>>, IEnumerable<KeyValuePair<byte[], byte[]>>, IEnumerable<KeyValuePair<int, int[]>>
+    {
+        private static readonly Encoding s_uTF8 = Encoding.UTF8;
+        public readonly static Encoding DefaultEncoding = s_uTF8;
+
+        private readonly Cache _cache;
+        private readonly Logger _infoLog;
+        private readonly Comparator _comparator;
+        private readonly Encoding _encoding;
+
+        static void Throw(IntPtr error)
+        {
+            Throw(error, msg => new Exception(msg));
+        }
+
+        static void Throw(IntPtr error, Func<string, Exception> exception)
+        {
+            if (error != IntPtr.Zero)
+            {
+                try
+                {
+                    var msg = Marshal.PtrToStringAnsi(error);
+                    throw exception(msg);
+                }
+                finally
+                {
+                    LevelDBInterop.leveldb_free(error);
+                }
+            }
+        }
+
+        public DB(Options options, string name)
+            : this(options, name, DefaultEncoding)
+        {
+        }
+
+        /// <summary>
+        /// Open the database with the specified "name".
+        /// </summary>
+        public DB(Options options, string name, Encoding encoding)
+        {
+            this._cache = options.Cache;
+            this._infoLog = options.InfoLog;
+            this._comparator = options.Comparator;
+            this.Handle = LevelDBInterop.leveldb_open(options.Handle, name, out var error);
+            this._encoding = encoding;
+
+            Throw(error, msg => new UnauthorizedAccessException(msg));
```

Comment:
Previously we call `FilterPolicy = Native.leveldb_filterpolicy_create_bloom(15)`

Reply:
What does that do or even mean for `leveldb`?

Reply:
Fixed add it, and it set bloom to `10` should need no more than `10` bits.